specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-12-01/machineLearningServices.json

```diff
@@ -0,0 +1,1956 @@
+        "maxNodeCount"
+      ]
+    },
+    "NodeStateCounts": {
+      "properties": {
+        "idleNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Idle node count.",
+          "description": "Number of compute nodes in idle state."
+        },
+        "runningNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Running node count.",
+          "description": "Number of compute nodes which are running jobs."
+        },
+        "preparingNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Preparing node count.",
+          "description": "Number of compute nodes which are being prepared."
+        },
+        "unusableNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Unusable node count.",
+          "description": "Number of compute nodes which are in unusable state."
+        },
+        "leavingNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Leaving node count.",
+          "description": "Number of compute nodes which are leaving the amlCompute."
+        },
+        "preemptedNodeCount": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "title": "Preempted node count.",
+          "description": "Number of compute nodes which are in preempted state."
+        }
+      },
+      "description": "Counts of various compute node states on the amlCompute."
+    },
+    "ClusterUpdateProperties": {
+      "properties": {
+        "scaleSettings": {
+          "$ref": "#/definitions/ScaleSettings",
+          "title": "Scale settings.",
+          "description": "Desired scale settings for the amlCompute."
+        }
+      },
+      "description": "The properties of a amlCompute that need to be updated."
+    },
+    "ClusterUpdateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ClusterUpdateProperties",
+          "description": "The properties of the amlCompute."
+        }
+      },
+      "description": "AmlCompute update parameters."
+    },
+    "VirtualMachineSshCredentials": {
+      "type": "object",
+      "description": "Admin credentials for virtual machine",
+      "properties": {
+        "username": {
+          "description": "Username of admin account",
+          "type": "string"
+        },
+        "password": {
+          "description": "Password of admin account",
+          "type": "string"
+        },
+        "publicKeyData": {
+          "description": "Public key data",
+          "type": "string"
+        },
+        "privateKeyData": {
+          "description": "Private key data",
+          "type": "string"

```

Comment:
SSH PrivateKey is the secret and should not be set in the resource, e.g. see VM Compute key model [here](https://github.com/Azure/azure-rest-api-specs/blob/bc2b1356222a0bda2b5d9fd7fbad7483bfe1ec08/specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json#L5613)

Reply:
that is used only for attach (user is bringing their own VM), and is used for the initial connection, we do not store that key, and it does not get set on the resource.
There is a different api (the listKeys one), that users can use to retrieve an ssh key to connect to the machine, and the returned sshKey will be different that the one provided by the user during attach.